addons/web/static/src/webclient/settings_form_view/settings_form_compiler.js

```diff
@@ -103,34 +104,27 @@ function compileLabel(el, params) {
     // localized *after* the field.
     // We don't know yet if the label refers to a field or not.
     if (res.textContent && res.tagName !== "FormLabel") {
-        params.labels.push(res.textContent.trim());
         labelsWeak.set(res, { textContent: res.textContent });
-        highlightElement(res);
-    }
-    return res;
-}
-
-function compileGenericLabel(el, params) {
-    const res = this.compileGenericNode(el, params);
-    if (res.textContent) {
-        params.labels.push(res.textContent.trim());
-        highlightElement(res);
+        params.labels.push(...highlightElement(res));
```

Comment:
if the label is inside the setting, doesn't it work already thanks to the call line 56?